Fix for issue #6

gemorlith opened the pull request:
Added UnitPathfinding.cs (partial of Unit)
Updated way to find reachable tiles.
Added Tile passable and cost methods.
Updated way to calculate distance

gemorlith committed:
Fix for issue #6

Added UnitPathfinding.cs (partial of Unit)
Updated way to find reachable tiles.
Added Tile passable and cost methods.
Updated way to calculate distance

Ebilkill commented:
Tile.cost must be capitalized

Unit.GetWalkablePositions is no longer used or necessary, as this is done by ReachableTiles. As such, it should be deleted or marked deprecated somehow. 

Please add more inline comments in the code, explaining what the code does (mostly in UnitPathfinding.cs)

Tile's methods are not in alphabetical order; this is not *required*, although it is preferred. 
Another preference, although not a requirement, is to have exactly one empty line between a statement and a block. For instance, like:

```C#
Console.WriteLine("");

for(int i = 0; i < list.Count; ++i)
{
    Console.WriteLine(list[i]);
}

Console.WriteLine("");
```

gemorlith committed:
Small fixes

halfway the fixes for the merge

gemorlith committed:
codestyle fixes

added comments in pathfinding and tiles and alphabetical order in tiles

gemorlith commented:
should be fixed.

twijgers committed:
Extra code style fix

Change comments to be more grammatically accurate, as well as improving spelling, capitalization, etc.
